gossip/src/cluster_info.rs

```diff
@@ -1290,13 +1290,24 @@ impl ClusterInfo {
             .collect()
     }
 
-    pub fn get_node_version(&self, pubkey: &Pubkey) -> Option<solana_version::LegacyVersion2> {
+    pub fn get_node_version(&self, pubkey: &Pubkey) -> Option<solana_version::Version> {
         let gossip_crds = self.gossip.crds.read().unwrap();
-        if let Some(version) = gossip_crds.get::<&Version>(*pubkey) {
-            return Some(version.version.clone());
-        }
-        let version: &crds_value::LegacyVersion = gossip_crds.get(*pubkey)?;
-        Some(version.version.clone().into())
+        gossip_crds
+            .get::<&ContactInfo>(*pubkey)
+            .map(ContactInfo::version)
+            .cloned()
+            .or_else(|| {
+                gossip_crds
+                    .get::<&Version>(*pubkey)
+                    .map(|entry| entry.version.clone())
+                    .or_else(|| {
+                        gossip_crds
+                            .get::<&crds_value::LegacyVersion>(*pubkey)
+                            .map(|entry| entry.version.clone())
+                            .map(solana_version::LegacyVersion2::from)
+                    })
+                    .map(solana_version::Version::from)
+            })
```

Comment:
If we have `Version` as its own `CrdsData` value, why do we need it in `ContactInfo`? Just in case we receive a `ContactInfo` and not `Version`?

Reply:
I assume we try `ContactInfo` first since it's more likely to be in our crds than `Version`?

Reply:
Since the new contact-info embeds the version, the current `CrdsData::Version` is going to be deprecated and we would stop gossiping that.

Each CRDS value incurs some overhead (pubkey, wallclock, signature) so it is better to have one CRDS value instead of splitting some data into two CRDS values.

Reply:
ya was wondering if that was the plan but didn't know since there was already a `LegacyVersion`